static/js/L.WebGLOverlay.js

```diff
@@ -0,0 +1,185 @@
+
+/*
+  Generic WebGL Overlay for leaflet, John Tyree
+*/
+
+if(typeof(L) !== 'undefined') {
+
+L.WebGLOverlay = L.Class.extend({
+
+  options: {
+    initialized: false
+  },
+
+  initialize: function (options) {
+    L.setOptions(this, options);
+  },
+
+  onAdd: function (map) {
+    this._map = map;
+    this._co = L.canvasOverlay().addTo(map);
+    this._gl = this._co.canvas().getContext('experimental-webgl', { antialias: true });
+    this._co.drawing(this.draw.bind(this))
+            .params(this.options);
+    var glLayer = this._co;
+    var canvas = glLayer.canvas();
+
+    // FIXME: What is this? Assigning to a function?
+    glLayer.canvas.width = canvas.clientWidth;
+    glLayer.canvas.height = canvas.clientHeight;
+
+    var gl = this._gl;
+
+    // -- WebGl setup
+    var vertexShader = gl.createShader(gl.VERTEX_SHADER);
+    gl.shaderSource(vertexShader, document.getElementById('vshader').text);
+    gl.compileShader(vertexShader);
+
+    var fragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+    gl.shaderSource(fragmentShader, document.getElementById('fshader').text);
+    gl.compileShader(fragmentShader);
+
+    // link shaders to create our program
+    var program = gl.createProgram();
+    gl.attachShader(program, vertexShader);
+    gl.attachShader(program, fragmentShader);
+    gl.linkProgram(program);
+    gl.useProgram(program);
+
+    // look up the locations for the inputs to our shaders.
+    this._u_matLoc = gl.getUniformLocation(program, "u_matrix");
+    this._vertLoc = gl.getAttribLocation(program, "a_vertex");
+
+    // Set the matrix to some that makes 1 unit 1 pixel.
+    this._pixelsToWebGLMatrix = new Float32Array(16);
+    this._mapMatrix = new Float32Array(16);
+    this._pixelsToWebGLMatrix.set([2 / canvas.width,  0                , 0, 0,
+                                   0               , -2 / canvas.height, 0, 0,
+                                   0               ,  0                , 0, 0,
+                                  -1               ,  1                , 0, 1]);
+    gl.viewport(0, 0, canvas.width, canvas.height);
+    gl.uniformMatrix4fv(this._u_matLoc, false, this._pixelsToWebGLMatrix);
+    this.initialized = false;
+  },
+
+  addTo: function (map) {
+    map.addLayer(this);
+    return this;
+  },
+
+  onRemove: function (map) {
+    this._co.onRemove(map);
+    this._gl = null;
+    this._map = null;
+    // FIXME: unintialize gl or someting?
+  },
+
+  LatLongToPixelXY: function(latitude, longitude) {
```

Comment:
I tried that first, actually. For some reason it was very, very slow by comparison.